feature-based-molecular-networking/tools/feature-based-molecular-networking/scripts/progenesis_formatter.py

```diff
@@ -147,17 +150,36 @@ def convert_mgf(input_msp, output_mgf, compound_to_scan_mapping):
 
             read_name = True
             scan = (compound_to_scan_mapping[compound_name])
-        elif line.startswith("PrecursorMZ:"):
-            precursor_mz = (line.rstrip().replace("PrecursorMZ: ", ""))
-        elif line.startswith("Charge:"):
-            charge = (line.rstrip().replace("Charge: ", ""))
-        elif len(line.rstrip()) == 0 and read_name == True:
+
+        if line.startswith("PrecursorMZ:"):
+            precursor_mz = (line.rstrip().replace("PrecursorMZ: ", "PEPMASS="))
+        if line.startswith("Charge:"):
+            charge = []
+            charge = (line.rstrip().replace("Charge: ", "CHARGE="))
+            precursor_type = []
+        elif line.startswith("Precursor_type:"):
+            precursor_type = []
+            precursor_type = line.rstrip().replace("Precursor_type: ", "ION=")
+            charge = []
+        if len(line.rstrip()) == 0 and read_name == True:
             read_name = False
 
             output_filename.write("BEGIN IONS\n")
             output_filename.write("SCANS=%s\n" % (compound_to_scan_mapping[compound_name]))
+            output_filename.write("FEATURE_ID=%s\n" % (compound_to_scan_mapping[compound_name]))
             output_filename.write("MSLEVEL=2\n")
-            output_filename.write("CHARGE=%s\n" % (line.rstrip().replace("Charge: ", "")))
+            output_filename.write(str(precursor_mz)+"\n")
+            if type(precursor_type) == str:
+                output_filename.write(str(precursor_type)+"\n")
```

Comment:
There is no prefix for this?

Reply:
It is included in the object above

Reply:
Do we need to write ion into the MGF since we never read it?

Reply:
What is the harm ?
We write it out in the MGF output in MZmine

Reply:
Lets go ahead and name it IONMODE= instead of ION=, this is consistent to how we do it in the libraries:

https://ccms-ucsd.github.io/GNPSDocumentation/api/#all-public-library-specta-at-gnps-with-peaks-and-annotation-history

Reply:
Here we ION= is for adduct, like in the MGF from MZmine.
IONMODE= is for positive/negative.
You have an ADDUCT= in the reference library MGF.
I would rather keep it consistent with MZmine MGF because it is for FBMN.